newlib/libc/locale/newlocale.c

```diff
@@ -127,8 +130,12 @@ newlocale (int category_mask, const char *locale,
     return __get_C_locale ();
 
 #ifndef _MB_CAPABLE
-  _REENT_ERRNO(p) = EINVAL;
-  return NULL;
+  if (locale[0] != '\0')
```

Comment:
Yup, this looks good now that I understand what's going on. Thanks for your patience.